calyx-py/calyx/builder_util.py

```diff
@@ -0,0 +1,190 @@
+# pylint: disable=import-error
+import calyx.builder as cb
+
+
+def insert_adder(
+    comp: cb.ComponentBuilder,
+    cell,
+    port_l,
+    port_r,
+    ans_reg,
```

Comment:
We could make this optional and if the user doesn't provide the reg, we could make one and return it as well or something?

Reply:
I began to do this, but I've sorta stopped short. If we offer to create a register, we'll also have to give that register a name, and AFAIK that name needs to be unique. So then, in the spirit of the rest of the development, I'd have to ask the user for a name. 

The difference then comes down to:
- Asking the user for a register that they have created.
- Asking the user for a string _name_ for a register that we will create.

This is very wishy-washy reasoning, but I also worry that letting a helper function add an important thing like a register to a component may be a bit much. For now, helper functions only add cells that are (at least spiritually) local to the group that they are creating. Thoughts? :)

Reply:
So, the `ComponentBuilder` object should ensure that there are no name conflicts (this is the interface for `ir::Builder` in Rust). All the provided names in our builders should be name hints and not the actual names of the primitives. To get a particular primitive, you better use its `CellBuilder` handle or something.

Reply:
If we have to pass names into all methods like this then we won't be able to abstract the behavior of things.

Reply:
Gotcha, this is now in!